wordpress-mobile/WordPress-iOS
# Backend-driven Dynamic Cards filtering

Fixes #22315

## Related PR
- https://github.com/wordpress-mobile/WordPressKit-iOS/pull/674

## Descrption
This PR calls the updates dashboard cards endpoint passing the extra parameters (device_id, platform, build_number, marketing_version, identifier) and removes the local dynamic cards filtering logic that was based on the deleted `remote_feature_flag` response field.

## Test Instructions
- Sandbox your emulator with D133371-code (instructions paqN3M-5m-p2)
- Open `FeatureFlagRemote.swift` file.
- At line 86, change `self.marketingVersion` to `23.9`.
- **Verify** that the domain management card does not appear
- Open `FeatureFlagRemote.swift` file.
- At line 86, change `self.marketingVersion` to `24.0`.
- **Verify** that the domain management card appears

## Note 
Please remove the `Do Not Merge` label and merge after:
- the backend change is deployed D133371-code
- The related WordPressKit PR is merged.
- The `WordPressKit` pod on this PR is updated to point to `trunk`.

## Regression Notes
1. Potential unintended areas of impact
New params are passed to the `Dashboard Cards` endpoint, this API change could unintentionally return incorrect cards. So, the content of the following cards should be tested:
   1. Today's Stats
   2. Scheduled Posts
   3. Activity
   4. Drafts
   5. Pages

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

## PR submission checklist:
